internal/stanza/integration_test.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/open-telemetry/opentelemetry-log-collection/operator"
+	"github.com/open-telemetry/opentelemetry-log-collection/operator/builtin/transformer/noop"
+	"github.com/open-telemetry/opentelemetry-log-collection/pipeline"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+)
+
+func createNoopReceiver(workerCount int, nextConsumer consumer.Logs) (*receiver, error) {
+	emitter := NewLogEmitter(zap.NewNop().Sugar())
+	logAgent, err := agent.NewBuilder(zap.NewNop().Sugar()).
+		WithConfig(&agent.Config{Pipeline: pipeline.Config{
+			operator.Config{
+				Builder: noop.NewNoopOperatorConfig(""),
+			},
+		}}).
+		WithDefaultOutput(emitter).
+		Build()
+	if err != nil {
+		return nil, err
+	}
+
+	opts := []ConverterOption{
+		WithLogger(zap.NewNop()),
+	}
+
+	if workerCount > 0 {
+		opts = append(opts, WithWorkerCount(workerCount))
+	}
+
+	converter := NewConverter(opts...)
+
+	return &receiver{
+		id:        config.NewComponentID("testReceiver"),
+		agent:     logAgent,
+		emitter:   emitter,
+		consumer:  nextConsumer,
+		logger:    zap.NewNop(),
+		converter: converter,
+	}, nil
+}
+
+// BenchmarkEmitterToConsumer serves as a benchmark for entries going from the emitter to consumer,
+// which follows this path: emitter -> receiver -> converter -> receiver -> consumer
+func BenchmarkEmitterToConsumer(b *testing.B) {
+	const (
+		entryCount = 1_000_000
+		hostsCount = 4
+	)
+
+	var (
+		workerCounts = []int{1, 2, 4, 6, 8}
+		entries      = complexEntriesForNDifferentHosts(entryCount, hostsCount)
+	)
+
+	for _, wc := range workerCounts {
+		b.Run(fmt.Sprintf("worker_count=%d", wc), func(b *testing.B) {
+			consumer := &mockLogsConsumer{}
+			logsReceiver, err := createNoopReceiver(wc, consumer)
+			require.NoError(b, err)
+
+			err = logsReceiver.Start(context.Background(), componenttest.NewNopHost())
+			require.NoError(b, err)
+
+			b.ResetTimer()
+
+			for i := 0; i < b.N; i++ {
+				consumer.ResetReceivedCount()
+
+				go func() {
+					ctx := context.Background()
+					for _, e := range entries {
+						_ = logsReceiver.emitter.Process(ctx, e)
+					}
+				}()
+
+				require.Eventually(b,
+					func() bool {
+						return consumer.Received() == entryCount
+					},
+					30*time.Second, 5*time.Millisecond, "Did not receive all logs (only received %d)", consumer.Received(),
+				)
+			}
+		})
+	}
+}
+
+func TestEmitterToConsumer(t *testing.T) {
+	const (
+		entryCount  = 1_000
+		hostsCount  = 4
+		workerCount = 2
+	)
+
+	entries := complexEntriesForNDifferentHosts(entryCount, hostsCount)
+
+	consumer := &mockLogsConsumer{}
+	logsReceiver, err := createNoopReceiver(workerCount, consumer)
+	require.NoError(t, err)
+
+	err = logsReceiver.Start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	consumer.ResetReceivedCount()
```

Comment:
I think this is probably an artifact from the corresponding benchmark, but not necessary in this test.